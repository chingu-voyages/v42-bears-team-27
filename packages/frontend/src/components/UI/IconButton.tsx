/* eslint-disable react/jsx-props-no-spreading */
import type {
  HTMLAttributes,
  RefAttributes,
  ForwardRefExoticComponent,
} from 'react';
import { forwardRef } from 'react';
import type { ThemeUIStyleObject } from 'theme-ui';

export interface IconButtonProps extends HTMLAttributes<HTMLButtonElement> {
  /**
   * Size of the component (width/height)
   * @default 32
   */
  size?: number;
  sx?: ThemeUIStyleObject;
}

export const IconButton: ForwardRefExoticComponent<
IconButtonProps & RefAttributes<HTMLButtonElement>
> = forwardRef(({ children, size = 32, sx, ...rest }, ref) => (
  <button
    ref={ref}
    type="button"
    sx={{
      variant: 'text.label',
      width: size,
      height: size,
      bg: 'transparent',
      border: 'none',
      cursor: 'pointer',
      ...sx,
    }}
    {...rest}
  >
    {children}
  </button>
));

if (process.env.NODE_ENV !== 'production') {
  IconButton.displayName = 'IconButton';
}
