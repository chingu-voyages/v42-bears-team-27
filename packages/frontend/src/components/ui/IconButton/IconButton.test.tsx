import { render } from 'test-utils';

import { IconButton } from './IconButton';

describe('<IconButton />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<IconButton />);
      expect(container).toMatchSnapshot();
    });
  });
});
