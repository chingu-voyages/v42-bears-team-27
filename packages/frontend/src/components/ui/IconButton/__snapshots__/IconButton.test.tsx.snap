// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> Snapshots should match snapshot 1`] = `
.emotion-0 {
  color: var(--theme-ui-colors-text);
  font-family: "Nunito",sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  width: 32px;
  height: 32px;
  padding: 0;
  background-color: transparent;
  border: none;
  cursor: pointer;
}

<div>
  <button
    class="emotion-0"
    type="button"
  />
</div>
`;
